generate_slicer_json.py

```diff
@@ -123,9 +123,10 @@ def int_converter(obj):
         return int(obj)
 
 
-def make_json(df):
+def make_json(df, filename = "segmentation_category_type"):
     '''
     This function composes the file and saves it to JSON format in current working directory.
+    Optional parameter of filename, provides default name 'segmentation_category_type.json'.
```

Comment:
Normally, the concept of "filename" includes the so-called extension (in Unix, there's never been anything special about a dot somewhere in the filename, and anything following a dot). But here you force a `.json` to be added to the end.

That's not necessarily a bad choice, but it should be expressly stated. Alternatively, take the filename as is (which for Unix command-line is actually the far more common behavior).

Reply:
Without the `.json` if someone puts in eg., `file.xlsx`, it will generate an excel file instead of a JSON. I could just set up a split if the filename contains a "." and take the first part to avoid that. In which case, I wouldn't need the `.json` in the `open` call.

Reply:
So it deduces the output format from the file extension? How would it do that given that `json.dump()` is passed the file pointer, not the file name? Or perhaps I'm misunderstanding?

Reply:
I removed the `.json` from the filename in `json.dump()` and ran a test: 
1. `python generate_slicer_json_opt_input.py terminology.csv test_json` generated a file called `test_json.json`, while 
2. `python generate_slicer_json_opt_input.py terminology.csv test_json.xlsx` generated an excel file: `test_json.xlsx`.

Reply:
How did you check that it's an Excel file? I ran that same test, and while the file then has an extension of `.xlsx`, the content was still perfectly valid JSON:

```sh
$ python3 generate_slicer_json.py test_data/terminology_test.csv foo.xlsx
$ cat foo.xlsx | head -10
{
    "SegmentationCategoryTypeContextName": "Segmentation category and type",
    "@schema": "https://raw.githubusercontent.com/qiicr/dcmqi/master/doc/schemas/segment-context-schema.json#",
    "SegmentationCodes": {
        "Category": [
            {
                "CodeMeaning": "Head",
                "CodingSchemeDesignator": "UBERON",
                "showAnatomy": true,
                "cid": "7150",
$ file foo.xlsx 
foo.xlsx: JSON data
```

Reply:
I see; I got that now too. That makes more sense.

Reply:
> I removed the `.json` from the filename in `json.dump()` and ran a test:

Can you say how you did that? `json.dump()` doesn't get passed a filename but a file pointer:
https://github.com/Imageomics/slicerMorph_JSON_generator/blob/bd13e83f2454be8653fa28504115febe1d7349c9/generate_slicer_json.py#L142

>    My guess would be that it defaults to JSON, but that can be overwritten by passing a different file extension, since it opens a blank file (giving a file extension would tell it a type to generate before the dump) and deposits the contents in there.

But the file is opened using the `open()` call, which doesn't set any file type regardless of name or extension:
https://github.com/Imageomics/slicerMorph_JSON_generator/blob/bd13e83f2454be8653fa28504115febe1d7349c9/generate_slicer_json.py#L141

Reply:
Yes, I mistakenly put too much weight on Finder's recognition of filetype. I had expected it to generate `foo.xlsx.json` as I have seen created accidentally in the past.

Reply:
Ah yes indeed, Finder does make some guesses from file extension 😊 But those should generally not be trusted ...

Reply:
I think this shows I should probably split the input on a `.` if it exists to save the unfortunate soul that types in the wrong file extension from a headache?

Reply:
You might see if there is no extension and then append `.json` automatically. But if there is one, it's common behavior on the Unix command line to honor the user's choices.

I would choose to make no alterations if the user provides a filename, because there's already a default with the proper extension.